Utilities/bootstrap

```diff
@@ -369,20 +442,29 @@ def create_bootstrap_files(sandbox_path, args):
         # Write out the target build commands (we just name the command and node
         # the same).
 
-        # Write the Swift compile commands, if used.
-        if target.swift_sources:
-            target.write_swift_compile_commands(
+        # Write the compile commands, if used.
+        if target.sources:
+            target.write_compile_commands(
                 args, target_build_dir, module_dir, inc_dir, output, objects,
                 link_input_nodes, predecessor_node)
 
+
         # Form the command line to link.
         linked_libraries = []
-        if target.is_library:
+        if target.is_swift and target.is_library:
             link_output_path = os.path.join(lib_dir, "%s.a" % (target.name,))
             link_command = ['rm', '-f', pipes.quote(link_output_path), ';',
                             'ar', 'cr', pipes.quote(link_output_path)]
             link_command.append(' '.join(pipes.quote(o) for o in objects))
-        else:
+        elif target.is_c and target.is_library:
+            # We have to use shared library for interpolation between Swift and C so we can't use static library for C Targets.
+            link_output_path = os.path.join(lib_dir, target.name + g_shared_lib_ext)
+            link_command = ['rm', '-f', pipes.quote(link_output_path), ';',
```

Comment:
No need for the `rm` here. That is used with archive libraries just to avoid dealing with `ar`'s broken incremental archive update, but with an output file like this the linker will just overwrite it.